.github/workflows/r_linux.yml

```diff
@@ -0,0 +1,109 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Linux R/C++ bundled static build
+
+on:
+  # TODO: make these CRON? or maybe just push to master?
+  # schedule:
+  #   - cron: |
+  #       0 */12 * * *
+  push:
+    branches:
+      - master
+    paths:
+      - 'ci/**'
+      - 'cpp/**'
+      - 'r/**'
+  # TODO: when making cron/push only, remove pull_request (for testing only right?)
+  pull_request:
+    paths:
+      - 'ci/**'
+      - 'cpp/**'
+      - 'r/**'
+
+jobs:
+
+  r-hub:
+    name: R-hub ${{ matrix.r_image }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        # See https://github.com/r-hub/rhub-linux-builders
+        r_image: ["debian-gcc-devel", "ubuntu-gcc-release"]
+    env:
+      R_DOCKER_ORG: "rhub"
+      R_DOCKER_IMAGE: ${{ matrix.r_image }}
+    steps:
+      - name: Checkout Arrow
+        uses: actions/checkout@v1
+      - name: Docker Pull
+        shell: bash
+        run: |
+          docker-compose pull --ignore-pull-failures r-docker
+      - name: Docker Build
+        shell: bash
+        run: |
+          docker-compose build r-docker
```

Comment:
As I mentioned in other comments, `docker-compose build rhub-r` would be better, as indicated by your job name too.

Reply:
Responded below. In sum, not all images come from `rhub` so `rhub-r` isn't an appropriate name.